core/src/main/java/tc/oc/pgm/stats/StatType.java

```diff
@@ -1,43 +1,87 @@
 package tc.oc.pgm.stats;
 
 import static net.kyori.adventure.text.Component.translatable;
-import static net.kyori.adventure.text.format.NamedTextColor.*;
+import static net.kyori.adventure.text.format.NamedTextColor.GREEN;
+import static net.kyori.adventure.text.format.NamedTextColor.RED;
+import static net.kyori.adventure.text.format.NamedTextColor.YELLOW;
 import static tc.oc.pgm.stats.StatsMatchModule.damageComponent;
 import static tc.oc.pgm.util.text.NumberComponent.number;
 
 import java.util.Locale;
 import net.kyori.adventure.text.Component;
+import tc.oc.pgm.util.math.Formula;
 
-public enum StatType {
-  KILLS,
-  DEATHS,
-  ASSISTS,
-  KILL_STREAK,
-  BEST_KILL_STREAK,
-  KILL_DEATH_RATIO,
-  LONGEST_BOW_SHOT {
-    private final String blocks = key + ".blocks";
+public sealed interface StatType {
+  Component makeNumber(Number number);
+
+  Component buildMessage(Component valueComponent);
+
+  Component component(StatHolder stats);
+
+  enum BaseStatType implements StatType {
```

Comment:
`StatType.BaseStatType` is both very long, and redundant in its name, consider something shorter like:
```suggestion
  enum Builtin implements StatType {
```
so it simplifies to `StatType.Builtin`

Reply:
Sure